src/main/java/book/store/controller/BookController.java

```diff
@@ -0,0 +1,38 @@
+package book.store.controller;
+
+import book.store.dto.BookDto;
+import book.store.dto.CreateBookRequestDto;
+import book.store.exception.EntityNotFoundException;
+import book.store.service.BookService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("api/books")
+public class BookController {
+    private final BookService bookService;
+
+    @GetMapping
+    public List<BookDto> getAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public BookDto getBookById(@PathVariable Long id) {
+        return bookService.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book with id: " + id)
+        );
+    }
```

Comment:
This was not the correct comment, we should not return Optional from the Service layer.
![image](https://github.com/SosunovychB/Online-Book-Store/assets/20326858/8f7124ce-c855-4340-9a06-d6a569496a45)

Reply:
Ok, I will return as it was before